apps/meteor/tests/e2e/prune-messages.spec.ts

```diff
@@ -0,0 +1,192 @@
+			await settings.set('FileUpload_FileSystemPath', FILE_SYSTEM_PATHS.CHANGED);
+			await sendFileMessage(poHomeChannel, FILE_NAMES.FILE_2);
+
+			let downloadUrl: string;
+			await test.step('download succeeds', async () => {
+				const downloadPromise = poHomeChannel.page.waitForEvent('download', (d) => d.suggestedFilename() === FILE_NAMES.FILE_2);
+				await poHomeChannel.content.lastUserMessage.getByRole('link', { name: 'Download' }).click();
+				const download = await downloadPromise;
+				expect(await download.failure()).toBeNull();
+				downloadUrl = download.url();
+			});
+
+			await settings.set('FileUpload_FileSystemPath', FILE_SYSTEM_PATHS.TEMPORARY);
+			await pruneMessages(poHomeChannel, { filesOnly: true });
+
+			await test.step('download fails with status 403 (forbidden)', async () => {
+				const download = await fetch(downloadUrl);
+				expect(download.status).toBe(403);
+			});
+
+			await settings.set('FileUpload_FileSystemPath', FILE_SYSTEM_PATHS.CHANGED);
+
+			await test.step('download succeeds', async () => {
+				const downloadPromise = poHomeChannel.page.waitForEvent('download', (d) => d.suggestedFilename() === FILE_NAMES.FILE_2);
+				await poHomeChannel.content.lastUserMessage.getByRole('link', { name: 'Download' }).click();
+				const download = await downloadPromise;
+				expect(await download.failure()).toBeNull();
+				downloadUrl = download.url();
+			});
+
+			await pruneMessages(poHomeChannel, { filesOnly: true });
+
+			await test.step('download fails with status 404 (not found)', async () => {
+				const download = await fetch(downloadUrl);
+				expect(download.status).toBe(404);
+			});
+
+			await pruneMessages(poHomeChannel, { filesOnly: false });
+		});
+
+		// FIXME: This will fail in the afterAll step when trying to delete the channel
+		test.fail('delete channel with dangling files', async ({ settings }) => {
+			await settings.set('FileUpload_FileSystemPath', FILE_SYSTEM_PATHS.CHANGED);
+			await sendFileMessage(poHomeChannel, FILE_NAMES.FILE_3);
+			await settings.set('FileUpload_FileSystemPath', FILE_SYSTEM_PATHS.TEMPORARY);
+		});
+
+		// FIXME: This will fail because the chat.delete API will return 400 (ENOENT)
+		test.fail('delete message with dangling files', async ({ settings }) => {
+			await settings.set('FileUpload_FileSystemPath', FILE_SYSTEM_PATHS.CHANGED);
+			await sendFileMessage(poHomeChannel, FILE_NAMES.FILE_3);
+			await settings.set('FileUpload_FileSystemPath', FILE_SYSTEM_PATHS.TEMPORARY);
+			await poHomeChannel.page.getByLabel('Close').click();
+			await poHomeChannel.content.openLastMessageMenu();
+			await poHomeChannel.page.getByRole('menuitem', { name: 'Delete' }).click();
+			const modal = poHomeChannel.page.getByLabel('Are you sure?');
+			await modal.getByRole('button', { name: 'Yes, delete' }).click();
+			await expect(modal).not.toBeVisible();
+			await expect(poHomeChannel.content.lastUserMessage).not.toBeVisible();
+		});
+	});
```

Comment:
If we delete the message when `fs.unlink` fails for any reason, then we lose track of that file and won't be able to delete it properly.

Ideally, the message gets deleted, but the reference to the file remains in the database so pruning is still attempted later.

I forced `ENOENT` in this test suite by changing `FileUpload_FileSystemPath` as just an easy way to reproduce. There are many possible errors:

https://man7.org/linux/man-pages/man2/unlink.2.html#ERRORS